crates/sui-types/src/storage.rs

```diff
@@ -1141,3 +1142,26 @@ impl WriteStore for SingleCheckpointSharedInMemoryStore {
         self.0.insert_committee(new_committee)
     }
 }
+
+pub trait BackingStore:
```

Comment:
I do not know how to feel about those traits making it higher and higher in the abstractions layer.
I am reworking the VM/Adapter and its caches which I hope will clean up some of this, and I am not suggesting we do anything about it. I am just venting my lack of understanding of the separation between storage and the adapter.
And it's just to hear opinion and nothing else